proof/refine/AARCH64/ArchCSpace_I.thy

```diff
@@ -11,699 +11,145 @@ theory ArchCSpace_I
-  "isArchObjectCap (capMasterCap cap) = isArchObjectCap cap"
-  "isThreadCap (capMasterCap cap) = isThreadCap cap"
-  "isCNodeCap (capMasterCap cap) = isCNodeCap cap"
-  "isNotificationCap (capMasterCap cap) = isNotificationCap cap"
-  "isEndpointCap (capMasterCap cap) = isEndpointCap cap"
-  "isUntypedCap (capMasterCap cap) = isUntypedCap cap"
-  "isReplyCap (capMasterCap cap) = isReplyCap cap"
-  "isIRQControlCap (capMasterCap cap) = isIRQControlCap cap"
-  "isIRQHandlerCap (capMasterCap cap) = isIRQHandlerCap cap"
-  "isNullCap (capMasterCap cap) = isNullCap cap"
-  "isDomainCap (capMasterCap cap) = isDomainCap cap"
-  "isArchFrameCap (capMasterCap cap) = isArchFrameCap cap"
-  "isArchObjectCap cap \<Longrightarrow> isSGISignalCap (capCap (capMasterCap cap)) = isSGISignalCap (capCap cap)"
-  by (auto simp: isCap_simps capMasterCap_def
-           split: capability.split arch_capability.split)
+lemmas arch_capMasterCap_eqDs[dest!] = arch_capMasterCap_eqDs1 arch_capMasterCap_eqDs1[OF sym]
 
 lemma capUntypedSize_capBits:
   "capClass cap = PhysicalClass \<Longrightarrow> capUntypedSize cap = 2 ^ (capBits cap)"
-  apply (simp add: capUntypedSize_def objBits_simps'
-                   AARCH64_H.capUntypedSize_def bit_simps'
-            split: capability.splits arch_capability.splits
-                   zombie_type.splits)
-  apply fastforce
-  done
+  by (fastforce simp: global.capUntypedSize_def objBits_simps bit_simps' capUntypedSize_def
+                split: capability.splits arch_capability.splits zombie_type.splits)
 
 lemma sameRegionAs_def2:
  "sameRegionAs cap cap' = (\<lambda>cap cap'.
-     (cap = cap'
-          \<and> (\<not> isNullCap cap \<and> \<not> isZombie cap
-              \<and> \<not> isUntypedCap cap \<and> \<not> isArchFrameCap cap)
-          \<and> (\<not> isNullCap cap' \<and> \<not> isZombie cap'
-              \<and> \<not> isUntypedCap cap' \<and> \<not> isArchFrameCap cap'))
-      \<or> (capRange cap' \<noteq> {} \<and> capRange cap' \<subseteq> capRange cap
-                 \<and> (isUntypedCap cap \<or> (isArchFrameCap cap \<and> isArchFrameCap cap')))
-      \<or> (isIRQControlCap cap \<and> isIRQHandlerCap cap')
-      \<or> (isIRQControlCap cap \<and> isArchObjectCap cap' \<and> isSGISignalCap (capCap cap')))
-           (capMasterCap cap) (capMasterCap cap')"
+    (cap = cap'
+     \<and> (\<not> isNullCap cap \<and> \<not> isZombie cap
+        \<and> \<not> isUntypedCap cap \<and> \<not> isArchFrameCap cap)
+     \<and> (\<not> isNullCap cap' \<and> \<not> isZombie cap'
+        \<and> \<not> isUntypedCap cap' \<and> \<not> isArchFrameCap cap'))
+    \<or> (capRange cap' \<noteq> {} \<and> capRange cap' \<subseteq> capRange cap
+       \<and> (isUntypedCap cap \<or> (isArchFrameCap cap \<and> isArchFrameCap cap')))
+    \<or> (isIRQControlCap cap \<and> isIRQHandlerCap cap')
+    \<or> (isIRQControlCap cap \<and> isArchObjectCap cap' \<and> isSGISignalCap (capCap cap')))
```

Comment:
Thank you for improving the indentation but I think there's still some unnecessary brackets making this more confusing than it needs to be.

Reply:
Yeah, I've been ignoring most of them for that reason but it's hard to resist the urge to suggest fixing it when it's a lemma that you're touching already.

Reply:
- [ ] reduce parens

Reply:
Updated to reduce parens and re-intent on all arches.